### DISCUSSION TITLE
Net flow data in grafan

### DISCUSSION THREAD

**Discussion by shubhamnyaik** [grafana-cold-storage/agent#3247] [Q&A]
Hi There, 

We are trying to onboard netflow in grafana. 
Is there a way by which we can get netflow data in grafana in prometheus format. 

We have tried using netflow collector but it has a dependency of influxDB. 
We are currently using prometheus backed databases like cortex and mimir. 
Please let me know if there is exporter/integration available for getting data in grafana in prometheus format.

There is already an integration/exporter for SNMP. We are looking for similar integration/exporter for netflow. 

Regards
Shubham

**Comment by tpaschalis** [comment]
I'm converting this issue to a Github discussion as it's not directly related to Grafana Agent, let's continue there.

**Comment by tpaschalis** [comment]
Hey there! The Grafana Agent does not have a built-in Netflow exporter.

As far as I understand, Netflow is a continuous stream of events and cannot be directly mapped to distinct metrics, but need to have some conversion pipeline to generate distinct metrics from them.

This needs a separate pipeline, for example using [goflow](https://github.com/cloudflare/goflow) to serialize and write these netflow event in Kafka, and then have [flow-exporter](https://github.com/neptune-networks/flow-exporter) read these Kafka events and emit Prometheus metrics that the Agent can scrape.

You could also take a look at [bio-routing/tflow2](https://github.com/bio-routing/tflow2) which does the same in-memory, but wouldn't know how well it works.

In any case, any PRs or contributions are welcome!
